plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/LibraryFactory.java

```diff
@@ -99,29 +105,64 @@ static Library create(IConfigurationElement configurationElement) throws Library
   }
 
   private static List<LibraryFile> getLibraryFiles(IConfigurationElement[] children)
-      throws InvalidRegistryObjectException, URISyntaxException {
+      throws InvalidRegistryObjectException, URISyntaxException, LibraryFactoryException {
     List<LibraryFile> libraryFiles = new ArrayList<>();
     for (IConfigurationElement libraryFileElement : children) {
       if (ELEMENT_NAME_LIBRARY_FILE.equals(libraryFileElement.getName())) {
         MavenCoordinates mavenCoordinates = getMavenCoordinates(
             libraryFileElement.getChildren(ELEMENT_NAME_MAVEN_COORDINATES));
-        LibraryFile libraryFile = new LibraryFile(mavenCoordinates);
-        libraryFile.setFilters(getFilters(libraryFileElement.getChildren()));
-        // todo do we really want these next two to be required?
-        libraryFile.setSourceUri(
-            getUri(libraryFileElement.getAttribute(ATTRIBUTE_NAME_SOURCE_URI)));
-        libraryFile.setJavadocUri(
-            getUri(libraryFileElement.getAttribute(ATTRIBUTE_NAME_JAVADOC_URI)));
-        String exportString = libraryFileElement.getAttribute(ATTRIBUTE_NAME_EXPORT);
-        if (exportString != null) {
-          libraryFile.setExport(Boolean.parseBoolean(exportString));
+        String loadDependencies = libraryFileElement.getAttribute("loadDependencies");
+        if ("true".equalsIgnoreCase(loadDependencies)) {
+          try {
+            Collection<LibraryFile> files = loadTransitiveDependencies(mavenCoordinates);
+            libraryFiles.addAll(files);
+          } catch (CoreException ex) {
+            throw new LibraryFactoryException(
+                "Problem loading transitive dependencies for " + mavenCoordinates, ex);
```

Comment:
I wonder if this should be externalized?

Reply:
Likely once it lands in its final location. I wonder, do we need some UI for surfacing this to the end user?